client_test/directdefund_integration_test.go

```diff
@@ -27,10 +28,31 @@ func TestDirectDefundIntegration(t *testing.T) {
 	chain := chainservice.NewMockChain()
 	broker := messageservice.NewBroker()
 
-	clientA, _ := setupClient(alice.PrivateKey, chain, broker, logDestination, 0)
-	clientB, _ := setupClient(bob.PrivateKey, chain, broker, logDestination, 0)
+	clientA, storeA := setupClient(alice.PrivateKey, chain, broker, logDestination, 0)
+	clientB, storeB := setupClient(bob.PrivateKey, chain, broker, logDestination, 0)
 
 	channelId := directlyFundALedgerChannel(t, clientA, clientB)
 	directlyDefundALedgerChannel(t, clientA, clientB, channelId)
 
+	// Ensure that we no longer have a consensus channel in the store
+	// And that we have a regular Channel instead
+	for _, store := range []store.Store{storeA, storeB} {
+
+		c, channelInStore := store.GetChannelById(channelId)
+
+		if !channelInStore {
+			t.Fatalf("expected a Channel to have been created")
+		}
+
+		if c.OnChainFunding.IsNonZero() {
+			t.Fatal("Expected zero on chain funding, but got nonzero")
+		}
+
+		_, err := store.GetConsensusChannelById(channelId)
+		if consensusChannelStillInStore := (err == nil); consensusChannelStillInStore {
+			t.Fatalf("Expected ConsensusChannel to have been destroyed in %v's store, but it was not", store.GetAddress())
+		}
+
+	}
```

Comment:
large diff here because the loop identifier `store` blocked access to `store.ErrNoSuchChannel`.

```suggestion
	for _, clientStore := range []store.Store{storeA, storeB} {

		c, channelInStore := clientStore.GetChannelById(channelId)

		if !channelInStore {
			t.Fatalf("expected a Channel to have been created")
		}

		if c.OnChainFunding.IsNonZero() {
			t.Fatal("Expected zero on chain funding, but got nonzero")
		}

		_, err := clientStore.GetConsensusChannelById(channelId)
		if err != store.ErrNoSuchChannel {
			t.Fatalf("Expected ConsensusChannel to have been destroyed in %v's store, but it was not", clientStore.GetAddress())
		}

	}
```